### ISSUE TITLE
Add support for overloaded arguments

### ISSUE THREAD

**Issue by rosshadden** [tj/dox#96]
The way I write many of my functions, the number and type of arguments are dynamic, allowing for developer-friendly (and developer-forgiving) APIs.

I see how you are dealing with this yourself, [in page.js for example](https://github.com/visionmedia/page.js/blob/master/index.js#L33), but I don't think that markup really conveys the point that the arguments can change nor that they can be infinite.

I realize that for one thing this is likely an edge case (though for me it is an every-day occurrence).  I realize, too, that this project is trying to adhere as much as possible to existing conventions like JSDoc.  And despite what [one of the creators of JSDoc said three years ago](http://www.thebrightlines.com/2010/03/28/document-your-javascript-interview-with-michael-mathews-of-jsdoc/) I am not sure that practice is even supported by it.

TL;DR:  Do you have any ideas on if/how we should support such a coding convention, which occurs most often in JavaScript libraries?
